EIPS/eip-7756.md

```diff
@@ -0,0 +1,177 @@
+---
+eip: 7756
+title: EOF/EVM Trace Specification
+description: Updates EIP-3155 tracing to add EOF support
+author: Martin Holst Swende (@holiman), Marius van der Wijden (@MariusVanDerWijden), Danno Ferrin (@shemnon)
+discussions-to: https://ethereum-magicians.org/t/eip-7756-eof-evm-trace-specification/20806
+status: Draft
+type: Standards Track
+category: Interface
+created: 2024-08-13
+requires: 3155
+---
+
+## Abstract
+
+Updates the [EIP-3155](./eip-3155.md) JSON tracing specification to support EOF features and light tracing.
+
+## Motivation
+
+EIP-3155 defined a tracing standard for Legacy EVM operations. However, the EVM Object Format ([EIP-7692](./eip-7692.md)) adds a number of features that need to be reflected in debugging traces.
+
+The use of these traces has also moved out from state testing, including live block tracing and differential fuzzing, increasing the need to keep tracing up to date.
+
+This EIP has multiple goals:
+
+- Add members to the trace object to support new EOF features.
+- Support tracing contracts contained in an EOF container as well as uncontained "legacy" contracts in the same trace.
+- Clarify any previous ambiguities in the EIP-3155 specification.
+
+## Specification
+
+To promote clarity and provide a cohesive specification, the entire tracing specification will be presented with alterations in-line rather than as a set of diffs on top of EIP-3155. Differences will be highlighted in the Backwards Compatibility section.
+
+### Datatypes
+
+| Type       | Explanation                                                    | Example                              |
+|------------|----------------------------------------------------------------|--------------------------------------|
+| Number     | Plain json number                                              | "pc":0                               |
+| Hex-Number | Hex-encoded number                                             | "gas":"0x2540be400"                  |
+| String     | Plain string                                                   | "opName":"PUSH1"                     |
+| Hex-String | Hex-encoded string                                             | "returnData":"0x60616203"            |
+| Array      | Array of Hex-Strings                                           | "stack":["0x0", "0x0"]               |
+| Key-Value  | Key-Value structure with key and values encoded as hex strings | "storage":{"0x0":"0x1", "0x1":"0x1"} |
+| Boolean    | Json bool can either be true or false                          | "pass": true                         |
```

Comment:
These examples are a little unclear. Is the "Plain json number" in the example column:

- the zero
- or the entire key/value pair where the value is zero